YeelightAPI/DeviceLocator.cs

```diff
@@ -301,15 +301,15 @@ private static List<Task<List<Device>>> CreateDiscoverTasks(NetworkInterface net
               .Select(
                 async networkInterface => await DeviceLocator.SearchNetworkForDevicesAsync(networkInterface, deviceFoundReporter, cancellationToken));
 
-            DiscoveryResult[] result = await Task.WhenAll(tasks);
+            DiscoveryResult[] results = await Task.WhenAll(tasks);
 
             //no results and exceptions has occured
-            if (result.All(r => !r.HasResult && r.HasError))
+            if (results.All(result => !result.HasResult && result.HasError))
```

Comment:
Please check if this condition is correct or the original condition is really wrong as I remarked. The original condition was discarding valid results.

Expected condition: throw if no devices were found and an exception was thrown.